src/cohortManager.js

```diff
@@ -0,0 +1,120 @@
+      this.cohorts.push(cohortToAdd)
+    } else {
+      return 'Cohort already exists'
+    }
+  }
+
+  searchCohort(name) {
+    const foundCohort = this.cohorts.find((cohort) => cohort.name === name)
+    if (foundCohort === undefined) return 'Cohort does not exist'
+    else return foundCohort
+  }
+
+  searchStudentById(id) {
+    const foundStudent = this.students.find((student) => student.id === id)
+    if (foundStudent === undefined) return 'Student does not exist'
+    return foundStudent
+  }
+
+  searchStudentByName(query) {
+    if (query.split(' ').length > 1) {
+      const [firstName, lastName] = query.toLowerCase().split(' ')
+      const res = []
+      this.students.forEach((student) =>
+        student.firstName.toLowerCase() === firstName &&
+        student.lastName.toLowerCase() === lastName
+          ? res.push(student)
+          : null
+      )
+      return res
+    } else {
+      const res = []
+      this.students.forEach((student) =>
+        student.firstName.toLowerCase() === query.toLowerCase() ||
+        student.lastName.toLowerCase() === query.toLowerCase()
+          ? res.push(student)
+          : null
+      )
+      if (res.length < 1) return 'No match found'
+      return res
+    }
+  }
+
+  newStudent(first, last, git, email) {
+    let nextId
+    if (this.students.length > 0) {
+      nextId = this.students[this.students.length - 1].id + 1
+    } else {
+      nextId = 1
+    }
+    const studentToAdd = new Student(nextId, first, last)
+    this.students.push(studentToAdd)
+  }
+
+  addStudentToCohort(id, cohort) {
+    const studentToAdd = this.searchStudentById(id)
+    const targetCohort = this.searchCohort(cohort)
+    if (studentToAdd === 'Student does not exist') return studentToAdd
+    if (targetCohort === 'Cohort does not exist') return targetCohort
+    const res = targetCohort.addStudent(studentToAdd)
+    if (res) return res
+    if (isTwilioEnabled) {
+      this.sendSMS(
+        `${studentToAdd.firstName} ${studentToAdd.lastName} just joined ${targetCohort.name} !`
+      )
+    }
+  }
+
+  removeCohort(name) {
+    const targetCohort = this.searchCohort(name)
+    if (targetCohort === 'Cohort does not exist') return targetCohort
+    const removeIndex = this.cohorts.findIndex(
+      (cohort) => cohort === targetCohort
+    )
+    this.cohorts.splice(removeIndex, 1)
+  }
+
+  removeStudentFromCohort(id, cohort) {
+    const studentToRemove = this.searchStudentById(id)
+    const targetCohort = this.searchCohort(cohort)
+    if (studentToRemove === 'Student does not exist') return studentToRemove
+    if (targetCohort === 'Cohort does not exist') return targetCohort
+    const res = targetCohort.removeStudent(studentToRemove)
+    if (res) return res
+    if (isTwilioEnabled) {
+      this.sendSMS(
+        `${studentToRemove.firstName} ${studentToRemove.lastName} just left ${targetCohort.name} !`
+      )
+    }
+  }
+
+  sendSMS(message) {
```

Comment:
you could extract this functionality out in a separate class that is responsible for delivering messages